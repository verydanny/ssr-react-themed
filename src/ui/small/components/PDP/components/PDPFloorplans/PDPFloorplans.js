import React from 'react'
import PropTypes from 'prop-types'
import { themed } from 'react-themed'
import cn from 'classnames'
import PDPSection from '../PDPSection'

const PDPFloorplans = ({
  theme,
  totalFloorplans,
  unitsAvailable,
}) => {
  const renderFloorplans = ({ expanded }) => {
    const listTheme = cn(
      theme.PDPFloorplans,
      { [theme['PDPFloorplans-expanded']]: expanded }
    )
    return ( // Return list and comparison placholder that will be built in a future story
      <div
        className={listTheme}
        data-tid="pdpFloorplansList"
      >
        <div>
            Floorplans<br /><br />
            Floorplans<br /><br />
            Floorplans<br /><br />
            Floorplans<br /><br />
            Floorplans<br /><br />
            Floorplans<br /><br />
        </div>
      </div>
    )
  }

  renderFloorplans.propTypes = { expanded: PropTypes.bool }

  return (
    <PDPSection
      category="Floorplans"
      dataTid="pdpFloorplans"
      description={`${totalFloorplans} floorplans and ${unitsAvailable} units available.`}
      expanded
      renderHidable={renderFloorplans}
      theme={theme}
    />
  )
}

PDPFloorplans.propTypes = {
  theme: PropTypes.object,
  totalFloorplans: PropTypes.number.isRequired,
  unitsAvailable: PropTypes.number.isRequired,
}

PDPFloorplans.defaultProps = {
  theme: {},
}

export default themed(/^PDPFloorplans/, { pure: true })(PDPFloorplans)
