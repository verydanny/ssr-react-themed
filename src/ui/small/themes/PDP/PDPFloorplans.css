@value (Gray4) from 'ui/shared/themes/colors/colors.css';

.PDPFloorplans {
  background-color: Gray4;
  display: none;
  padding-top: 21px;
}

.PDPFloorplans-expanded {
  display: block;
}
