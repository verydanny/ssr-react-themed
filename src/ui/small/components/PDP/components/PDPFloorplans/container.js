import { connect } from 'react-redux'
import { getTotalFloorplans, getUnitsAvailable } from 'app/selectors/pdpListing'
import PDPFloorplans from './PDPFloorplans'

const mapStateToProps = state => ({
  totalFloorplans: getTotalFloorplans(state),
  unitsAvailable: getUnitsAvailable(state),
})

export default connect(mapStateToProps, {})(PDPFloorplans)
