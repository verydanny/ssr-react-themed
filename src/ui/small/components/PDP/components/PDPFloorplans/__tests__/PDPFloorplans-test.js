import React from 'react'
import { shallow, mount } from 'enzyme'
import PDPFloorplans from '../PDPFloorplans'

const defaultProps = {}

const standardProps = {
  totalFloorplans: 10,
  unitsAvailable: 50,
}

const descriptionText = '10 floorplans and 50 units available.'

const factory = props =>
  <PDPFloorplans.WrappedComponent {...defaultProps} {...props} />

describe('<PDPFloorplans />', () => {
  it('should render', () => {
    const wrapper = shallow(factory(standardProps))

    expect(wrapper.exists()).toBe(true)
  })

  it('description should render correctly with totals', () => {
    const wrapper = mount(factory(standardProps))
    const description = wrapper.find('span[data-tid="pdpFloorplans_Description"]')
    expect(description.text()).toEqual(descriptionText)
  })

  it('render should match snapshot', () => {
    const wrapper = shallow(factory(standardProps))
    expect(wrapper.html()).toMatchSnapshot()
  })
})
