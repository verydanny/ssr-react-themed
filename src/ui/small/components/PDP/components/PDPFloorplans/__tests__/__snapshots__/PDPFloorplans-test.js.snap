// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PDPFloorplans /> render should match snapshot 1`] = `

<section data-tid="pdpFloorplans">
  <div role="Button"
       tabindex="0"
       data-tid="pdpFloorplans_Container"
       class="undefined"
  >
    <div>
      <h2 data-tid="pdpFloorplans_Header">
        Floorplans
      </h2>
      <button data-tid="pdpFloorplans_Button"
              class="undefined"
      >
      </button>
    </div>
    <div>
      <span data-tid="pdpFloorplans_Description">
        10 floorplans and 50 units available.
      </span>
    </div>
  </div>
  <div>
    <div class="undefined"
         data-tid="pdpFloorplansList"
    >
      <div>
        Floorplans
        <br>
        <br>
        Floorplans
        <br>
        <br>
        Floorplans
        <br>
        <br>
        Floorplans
        <br>
        <br>
        Floorplans
        <br>
        <br>
        Floorplans
        <br>
        <br>
      </div>
    </div>
  </div>
</section>

`;
